packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2618,7 +2641,7 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
 
   @override
   void paint(PaintingContext context, Offset offset) {
-    if (_hoverAnnotation != null) {
+    if (_hoverAnnotation != null && RendererBinding.instance.mouseTracker.mouseIsConnected) {
```

Comment:
Can you update needsCompositing as well? Compositing is only necessary if we push a layer, and this changes when we push layers.

(Might be worth factoring the condition out into a common place so we don't miss this again next time).

Reply:
OK, done.